lightning/src/events/mod.rs

```diff
@@ -79,6 +79,28 @@ impl_writeable_tlv_based_enum!(PaymentPurpose,
 	(2, SpontaneousPayment)
 );
 
+/// Information about an HTLC that is part of a payment that can be claimed.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ClaimedHTLC {
+	/// The SCID over which the HTLC was received.
+	pub short_channel_id: u64,
+	/// The HTLC's ID.
+	pub htlc_id: u64,
+	/// The block height at which this HTLC expires.
+	pub cltv_expiry: u32,
+	/// The amount (in msats) of this part of an MPP.
+	pub value: u64,
+	/// The sender-indented sum total of all MPP parts.
+	pub total_msat: u64,
```

Comment:
This field should probably be set in the top-level `PaymentClaimed` event since it should be required to be the same for all HTLCs. Maybe we could follow up adding this field? Also I think it should be called `sender_intended_total_msat`

Reply:
I just went ahead and did this... look ok?